### PR TITLE
Feature/20300 pupilspa webspeech api

### PR DESCRIPTION
**Pull request by danwilliams**

### CODE REVIEW

**Comment by GuyHarwood** [comment]
looks like you have test failures in the pupil app...

File: pupil-spa/src/app/check-complete/check-complete.component.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+export class CheckCompleteComponent implements OnInit, AfterViewInit, OnDestroy {
 
   protected window: any;
 
-  constructor(private storageService: StorageService, protected windowRefService: WindowRefService) {
+  constructor(private storageService: StorageService,
+              protected windowRefService: WindowRefService,
+              private questionService: QuestionService,
+              private speechService: SpeechService,
+              private elRef: ElementRef) {
```
Comment (jon-shipley) [inline pupil-spa/src/app/check-complete/check-complete.component.ts]: 👍 elegantly simple

File: pupil-spa/src/app/check-complete/check-complete.component.ts [hunk 2]
```diff
@@ -23,4 +29,15 @@ export class CheckCompleteComponent implements OnInit {
     });
   }
 
+  // wait for the component to be rendered first, before parsing the text
+  ngAfterViewInit() {
+    if (this.questionService.getConfig().speechSynthesis) {
+      this.speechService.speakElement(this.elRef.nativeElement);
```
Comment (jon-shipley) [inline pupil-spa/src/app/check-complete/check-complete.component.ts]: nice

**Comment by jon-shipley** [comment]
@danwilliams - please address the failing unit test

**Comment by danwilliams** [comment]
Okay, the failing unit tests were a little odd. They all worked fine for me when I submitted for PR, but I hadn't checked against the latest master (my bad!). Turns out Jon had removed some lines from `pupil-spa/src/app/question/question.component.spec.ts` as part of 20299. I have added them back in to fix the issue, and the tests now pass again.

**Pull request merged**